### DISCUSSION TITLE
Manager APIMAN - KEYCLOAK : Logout

### DISCUSSION THREAD

**Discussion by JYassine** [apiman/apiman#1767] [Help]
**SETUP**

Apiman version : 2.2.0.Final

Keycloak version : 15.1.1

Environment

Operating system : Linux
Java 11
Any custom JVM settings ? No
Manager platform : Wildfly

Gateway platform : Wildfly

**Description :**
Hello,

We would like to have more information on APIMAN and keycloak exchange because we ran into a problem with the logout of the manager APIMAN.

**Issue #1757 fixed our authentication (login) problem, but the APIMAN manager logout no longer works.** 

**Simplified architecture of our application :**
reverse-proxy <> APIMAN manager (keycloak on the same server).

**The process is as follows :**

When you logout to the APIMAN manager interface (API configuration administration console), the following error appears :

**"failed to invoke remote logout: java.net.ConnectException: Connection timed out (Connection timed out)"**

**Screenshots**

![image](https://user-images.githubusercontent.com/44097638/150750783-7436fcdf-82c4-4190-9e8a-c45acc6a110e.png)

How can we configure the keycloak so that the client can logout in the manager APIMAN ? 

**Test :**

We tested to change the configuration "Valid Redirect URIs" (apimanui client) in the keycloak :

Attribute : Valid Redirect URIs

In the description of keycloak of this attribute, it is the same URL that is used for login and logout.

**Screenshots**
![image](https://user-images.githubusercontent.com/44097638/150750818-822d5d6a-5756-4dbc-a84e-d95347d1796f.png)

**Expected behaviour**

When we logout it redirect to authentication page of APIMAN manager.

Could you please help us on this matter?

Thank you for your help and collaboration,
Sincerely.
Jrad Yassine

**Comment by msavy** [comment]
Good evening, @JYassine.

Your first problem wasn't a bug; it was a configuration issue with Keycloak (hence why it was moved to discussions). [If you have a moment, can you please also mark which reply helped fix your problem](https://github.com/apiman/apiman/discussions/1758), this will ensure other community users with the same issue can benefit. 

From what I can see, this exception you are seeing is also a Keycloak configuration issue. Community-grade support here is generally focussed on Apiman only, and is only provided when contributors (such as me) have spare time available to reply.

NB: please provide stack traces as blocks or [gist](gist.github.com), not as screenshots. Screenshots are much harder/impossible to analyse.
